pkg/authorization/apis/authorization/types.go

```diff
@@ -626,3 +626,66 @@ type ServiceAccountReference struct {
 	// which the ServiceAccountReference is embedded is used.
 	Namespace string
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AccessRestriction is used to guard specific actions without invasive changes to the cluster's default RBAC policy.
+// It supports either a whitelist or a blacklist based restriction.  It never grants any privileges - it can only be
+// used to take privileges away.
+type AccessRestriction struct {
+	metav1.TypeMeta
+
+	// Standard object's metadata.
+	metav1.ObjectMeta
+
+	// Spec defines when this restriction is imposed and how to satisfy it.
+	Spec AccessRestrictionSpec
+}
+
+// AccessRestrictionSpec holds the matching requirements.
+// MatchAttributes is required.
+// One of AllowedSubjects or DeniedSubjects must be specified.
+type AccessRestrictionSpec struct {
+	// If these rules cover the current request, then this restriction applies.
+	// If AllowedSubjects is set, then only those subjects can perform the matching actions.
+	// If DeniedSubjects is set, then only those subjects are restricted from performing the matching actions.
+	// Required.
+	MatchAttributes []rbac.PolicyRule
```

Comment:
FYI: this'll need to be `rbacapi.PolicyRule` when you rebase with latest master (where [e0243fb](https://github.com/openshift/origin/commit/e0243fb0964962a2d30e7372152f2b4e3ee40245#diff-e2a0970231b1593db088a5e7e353c468L44) removed the duplicate import)